src/core/memory.cpp

```diff
@@ -67,6 +67,16 @@ struct Memory::Impl {
         return system.DeviceMemory().GetPointer(paddr) + vaddr;
     }
 
+    [[nodiscard]] u8* GetPointerFromDebugMemory(VAddr vaddr) const {
+        const PAddr paddr{current_page_table->backing_addr[vaddr >> PAGE_BITS]};
+
+        if (!paddr) {
```

Comment:
For arithmetic types, prefer making the comparison explicit:

```suggestion
        if (paddr == 0) {
```